src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    if (this.health <= 0) {
+      const animalDiedIndex = Animal.alive.indexOf(this);
+
+      if (animalDiedIndex !== -1) {
+        Animal.alive.splice(animalDiedIndex, 1);
+      }
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        animal.die();
+      }
+    }
```

Comment:
The `bite` method currently only allows `Carnivore` to bite `Herbivore` instances that are not hidden. If you plan to extend the functionality to other types of animals, consider making this method more generic. For instance, you could check if the `animal` has a `health` property instead of specifically checking for `Herbivore` and `hidden`.